proj/funcDictionary.py

```diff
@@ -0,0 +1,335 @@
+        clustersizesAllDifSquare,(lAll - 1)))
+    plotData.diametersAllErr = np.sqrt(np.divide( \
+        diametersAllDifSquare,(lAll - 1)))
+    #centersAllSumSquare = np.sqrt(np.divide( \
+    #    centersAllSumSquare,(lAll - 1)))
+    #eccentritiesAllSumSquare = np.sqrt(np.divide( \
+    #    eccentritiesAllSumSquare,(lAll - 1)))
+    plotData.degreeMeanAllErr = np.sqrt(np.divide( \
+        degreeMeanAllDifSquare,(lAll - 1)))
+    # Shouldn't we also set the Mean properties?
+
+    return plotData
+
+def make_all_data(plotData, innerData, sample): # check description
+    """
+    info: calculate the sum of the clusterproperties of innerData and 
+        plotData (thus the mean over all samples can be calculated later) # right? # Where is the normalization?
+    input: plotData: object, storing the network properties
+        innerData: object, storing the network properties for one sample # right?
+        sample: the current sample # right? Or better "samples" as input? Where is it used?
+    output: plotData: 
+    """
+
+    plotData.clustersizesAllArray.append(innerData.clustersizes)
+    plotData.diametersAllArray.append(innerData.diameters)
+    plotData.centersAllArray.append(innerData.centers)
+    plotData.eccentritiesAllArray.append(innerData.eccentrities)
+    plotData.degreeMeanAllArray.append(innerData.degreeMean)
+    
+    return plotData
+
+def convert_to_amino_acids(plotData, innerData): # check description
+    """
+    info: convert lengths to the aminoacids
+    input: plotData: PlotData object, that contains the important data and parameters;
+            for more detail see documentation in plotData.py
+        innerData: objects, that store network properties in list # in more detail perhaps
+    output: plotData: object, that stores the network properties, now with updated members
+    """
+
+    xArgsNew = list()
+    clustersizesNew = list()
+    diametersNew = list()
+    eccentritiesNew = list()
+    degreeMeanNew = list()
+
+    for i in range(len(innerData.xArgs)):
+        if i%3 == 0:
+            xArgsNew.append(innerData.xArgs[i])
+            clustersizesNew.append(innerData.clustersizes[i])
+            diametersNew.append(innerData.diameters[i])
+            eccentritiesNew.append(innerData.eccentrities[i])
+            degreeMeanNew.append(innerData.degreeMean[i])
+
+    plotData.clustersizesAllArray.append(clustersizesNew)
+    plotData.diametersAllArray.append(diametersNew)
+    #plotData.centersAllArray.append(centers)
+    plotData.eccentritiesAllArray.append(eccentritiesNew)
+    plotData.degreeMeanAllArray.append(degreeMeanNew)
+    plotData.xArgsAllArray.append(xArgsNew)
+    
+    return plotData # that belongs here, right?
+
+def degree_mean(GSub):
+    """
+    info: calculate the average degree in the graph GSub
+    input: GSub: graph
+    output: deg: average degree number (real number)
+    """
+    deg = 0
+    degs = GSub.degree()
```

Comment:
This is super inefficient. Just use something like `sum(degrees)/degrees.size` or  `sum(degrees)/len(degrees)`. Try to avoid `for` loops whenever you can and use built-in functions such as `numpy.sum()`, etc.